### PR TITLE
Disable optimize_rewrite_sum_if_to_count_if by default

### PR DESCRIPTION
**Pull request by den-crane**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Disables optimize_rewrite_sum_if_to_count_if by default, mitigates: https://github.com/ClickHouse/ClickHouse/issues/38605 https://github.com/ClickHouse/ClickHouse/issues/38683

### CODE REVIEW

File: tests/queries/0_stateless/01646_rewrite_sum_if_bug.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+DROP TABLE IF EXISTS t;
+create table t( s String ) Engine=Memory as select arrayJoin (['a','b','c']);
+
+SELECT round((sum(multiIf(s IN ('a', 'b'), 1, 0)) / count()) * 100) AS r
+FROM cluster('test_cluster_two_shards', currentDatabase(), t);
+
+DROP TABLE t;
```
Comment (CurtizJ) [inline tests/queries/0_stateless/01646_rewrite_sum_if_bug.sql:7]: BTW, this case is fixed in master.

File: tests/queries/0_stateless/01646_rewrite_sum_if_bug.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+WITH 
+    sum(if((a >= 0) AND (b != 100) AND (c = 0), 1, 0)) AS r1, 
+    sum(if((a >= 0) AND (b != 100) AND (c > 220), 1, 0)) AS r2 
+SELECT 
+    (intDiv(toUInt32(rtime), 20) * 20) * 1000 AS t, 
+    (r1 * 100) / (r1 + r2) AS m
+FROM cluster('test_cluster_two_shards', currentDatabase(), test_alias)
+WHERE day = '2022-01-01'
+GROUP BY t
+ORDER BY t ASC;
```
Comment (CurtizJ) [inline tests/queries/0_stateless/01646_rewrite_sum_if_bug.sql:24]: But this is not :(

**Review by CurtizJ — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@CurtizJ What was the cause of this issue and why disabling the setting required?

**Comment by CurtizJ** [comment]
> What was the cause of this issue
I don't know, maybe wrong analysis of `WITH` clause in dustributed queries.

> why disabling the setting required
The issue was opened more than 2 months ago, nobody wants to fix it. As I understood after merging of new analyzer AST optimization should be deprecated and rewritten to query tree optimizations.

**Pull request merged**